include/boost/geometry/algorithms/detail/disjoint/areal_areal.hpp

```diff
@@ -129,6 +179,67 @@ struct areal_areal
     }
 };
 
+template <typename Areal, typename Box>
+struct areal_box_with_info
+{
+    typedef typename point_type<Areal>::type point_type;
+    typedef segment_intersection_points<point_type> intersection_return_type;
+
+    template <typename Strategy>
+    static inline intersection_return_type
+    apply(Areal const& areal,
+          Box const& box,
+          Strategy const& strategy)
+    {
+        intersection_return_type res =
+                for_each_segment(geometry::segments_begin(areal),
+                                 geometry::segments_end(areal),
+                                 box, strategy);
+                                 //strategy.get_disjoint_segment_box_strategy());
+        if ( res.count != 0 )
+        {
+            return res;
+        }
+
+        // If there is no intersection of any segment and box,
+        // the box might be located inside areal geometry
+
+        if ( point_on_border_covered_by(box, areal,
+                strategy.template get_point_in_geometry_strategy<Box, Areal>()) )
+        {
+            res.count = 1;
+            res.intersections[0] = box.max_corner();
```

Comment:
User-defined box may not define `max_corner()`.

Reply:
Ok, I will do it with `get()` and `set()`.
However it is not clear to me from the definition of the concept https://www.boost.org/doc/libs/1_73_0/libs/geometry/doc/html/geometry/reference/concepts/concept_box.html that user-defined box may not define `max_corner()`.

Reply:
It may or may not define it. The coordinates of a user-defined box or segment are accessed with 2-argument get/set by the library. The library can access the points with `max_corner` but only if it's one of the types we provide. Btw, if you need to access the point you can also use `indexed_point_view` (see: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/views/detail/indexed_point_view.hpp).